internal/dcache/metadata_manager/metadata_manager_impl.go

```diff
@@ -125,46 +161,209 @@ func GetClusterMap() ([]byte, *azcore.ETag, error) {
+		Metadata:               metadata,
+		Data:                   fileMetadata,
+		IsNoneMatchEtagEnabled: true,
+		EtagMatchConditions:    "",
+	})
+	if err != nil {
+		if bloberror.HasCode(err, bloberror.ConditionNotMet) {
+			log.Warn("CreateFileInit :: PutBlobInStorage for %s failed due to ETag mismatch", path)
+			return nil
+		}
+		log.Err("CreateFileInit :: Failed to put blob %s in storage: %v", path, err)
+	}
+	return err
 }
 
 // CreateFileFinalize finalizes the metadata for a file
-func (m *BlobMetadataManager) createFileFinalize(filePath string, fileMetadata *dcache.FileMetadata) error {
-	// Dummy implementation
-	return nil
+func (m *BlobMetadataManager) createFileFinalize(filePath string, fileMetadata []byte) error {
+	path := filepath.Join(m.mdRoot, "Objects", filePath)
+	// TODO :: check metadata property is not overwritten by this
+	err := m.storageCallback.PutBlobInStorage(internal.WriteFromBufferOptions{
+		Name:                   path,
+		Data:                   fileMetadata,
+		IsNoneMatchEtagEnabled: false,
+		EtagMatchConditions:    "",
+	})
+	if err != nil {
+		log.Err("CreateFileFinalize :: Failed to put blob %s in storage: %v", path, err)
+	}
+	return err
 }
 
 // GetFile reads and returns the content of metadata for a file
 func (m *BlobMetadataManager) getFile(filePath string) (*dcache.FileMetadata, error) {
-	// Dummy implementation
-	return nil, nil
+	path := filepath.Join(m.mdRoot, "Objects", filePath)
+	// Get the metadata content from storage
+	data, err := m.storageCallback.GetBlobFromStorage(internal.ReadFileWithNameOptions{
+		Path: path,
+	})
+	if err != nil {
+		log.Debug("GetFile :: Failed to get metadata file content for file %s : %v", path, err)
+		return nil, err
+	}
+	// Unmarshal the JSON data into the Metadata struct
+	var metadata dcache.FileMetadata
+	err = json.Unmarshal(data, &metadata)
+	if err != nil {
+		log.Debug("GetFile :: Failed to unmarshal JSON data: %v", err)
+		return nil, err
+	}
+	// Return the Metadata struct
+	return &metadata, nil
 }
 
 // DeleteFile removes metadata for a file
 func (m *BlobMetadataManager) deleteFile(filePath string) error {
-	// Dummy implementation
-	return nil
+	path := filepath.Join(m.mdRoot, "Objects", filePath)
+	err := m.storageCallback.DeleteBlobInStorage(internal.DeleteFileOptions{
+		Name: path,
+	})
+	if err != nil {
+		if bloberror.HasCode(err, bloberror.BlobNotFound) {
+			log.Warn("DeleteFile :: DeleteBlobInStorage failed since blob %s is already deleted", path)
+			return nil
+		}
+	}
+	return err
 }
 
 // OpenFile increments the open count for a file and returns the updated count
 func (m *BlobMetadataManager) openFile(filePath string) (int64, error) {
-	// Dummy implementation
-
-	return 0, nil
+	path := filepath.Join(m.mdRoot, "Objects", filePath)
+	count, err := m.updateHandleCount(path, true)
+	if err != nil {
+		log.Err("OpenFile :: Failed to update file open count for path %s : %v", path, err)
+		return -1, err
+	}
+	return count, nil
```

Comment:
add a debug log before this telling that file successfully opened, and the new opencount is

same for close

Reply:
// A file successfully opened, must have an opencount > 0.
common.Assert(count > 0, "Open file cannot have count <= 0", count)